snorkel/analysis/utils.py

```diff
@@ -0,0 +1,130 @@
+import numpy as np
+import scipy.sparse as sparse
+import torch
+
+from snorkel.types import ArrayLike
+
+
+def prob_to_pred(prob: np.ndarray) -> np.ndarray:
+    """Convert an array of probabilistic labels into an array of predictions
+
+    Parameters
+    ----------
+    prob
+        A [num_datapoints, num_classes] array of probabilistic labels such that each
+        row sums to 1.
+
+    Returns
+    -------
+    np.ndarray
+        A [num_datapoints, 1] array of predictions (integers in [1, ..., num_classes])
+    """
+    assert prob.ndim == 2
+    assert (prob <= 1).all()
```

Comment:
This is expensive, and if we do this, we should also check 0 <= p <= 1, p.sum(1) = 1. My vote is to remove and rely on contract in the docstring.